.github/actions/publish-nuget/action.yml

```diff
@@ -62,8 +62,8 @@ runs:
 
         if ($version -eq 'USE_GITHUB_REF_NAME') {
             $version = $Env:GITHUB_REF_NAME.Trim().TrimStart("v")
-            if (!($version -match '^\d+\.\d+\.\d+(-alpha\.\d+\.[a-z]{2,}-\d+)?$')) {
-                Write-Output "::error::`"v$version`" is invalid. The correct format is `"vM.N.O`" or `"vM.N.O-alpha.X.<issue-code>`"."
+            if (!($version -match '^\d+\.\d+\.\d+(-(beta|preview|alpha|rc)(\.\d+(\.[a-zA-Z]{2,}(-\d+)?)?)?)?$')) {
+                Write-Output "::error::`"v$version`" is invalid. The correct format is `"vM.N.O`" or `"vM.N.O-[alpha, preview , beta , rc].X.<issue-code>`"."
```

Comment:
```suggestion
                Write-Output "::error::`"v$version`" is invalid. The correct format is `"vM.N.O`" or `"vM.N.O-[alpha, preview , beta , rc]` with an optional `.X.<issue-code>` suffix"."
```

Reply:
```
$versionPattern = '^\d+\.\d+\.\d+(-(beta|preview|alpha|rc)(\.\d+(\.[a-zA-Z]{2,}(-\d+)?)?)?)?$'
$versions = @(
    '1.0.0',
    '2.0.0-beta',
    '3.0.0-rc.1',
    '4.0.0-preview.2.3',
    '5.0.0-alpha.1.issue-123'
)

foreach ($version in $versions) {
    if ($version -match $versionPattern) {
        Write-Host "Valid version: $version"
    } else {
        Write-Host "Invalid version: $version"
    }
}
```
Valid version: 1.0.0
Valid version: 2.0.0-beta
Valid version: 3.0.0-rc.1
Invalid version: 4.0.0-preview.2.3 _( because the issue code starts with letters and then numbers)_
Valid version: 5.0.0-alpha.1.issue-123